### PR TITLE
[prom-label-proxy] kube-rbac-proxy support

### PR DESCRIPTION
**Pull request by sathieu**
#### What this PR does / why we need it

Add kube-rbac-proxy support to prom-label-proxy.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by jkroepke** [comment]
Hi @sathieu 

could you please add a ci test values files like https://github.com/prometheus-community/helm-charts/blob/main/charts/jiralert/ci/test-values.yaml where a full blown configuration for rbac-proxy exists? Then, everything is covered inside CI

**Review by jkroepke — CHANGES_REQUESTED** [review]
Hi @sathieu

could you please add a ci test values files like [main/charts/jiralert/ci/test-values.yaml](https://github.com/prometheus-community/helm-charts/blob/main/charts/jiralert/ci/test-values.yaml?rgh-link-date=2024-06-29T15%3A30%3A45Z) where a full blown configuration for rbac-proxy exists? Then, everything is covered inside CI

**Comment by sathieu** [comment]
@jkroepke I've added `charts/prom-label-proxy/ci/kube-rbac-proxy-values.yaml`, back to you 🏀 !

File: charts/prom-label-proxy/templates/rbac-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ template "prom-label-proxy.fullname" . }}-rbac-config
+  namespace: {{ template "prom-label-proxy.namespace" . }}
+  labels: {{- include "prom-label-proxy.labels" . | nindent 4 }}
+  {{- if .Values.annotations }}
+  annotations:
+    {{ toYaml .Values.annotations | nindent 4 }}
+  {{- end }}
+data:
+  config-file.yaml: |{{ .Values.kubeRBACProxy.config | toYaml | nindent 4 }}
```
Comment (jkroepke) [inline charts/prom-label-proxy/templates/rbac-configmap.yaml:13]: tbh honest, I never saw this kind of style where is pipe is inline. Pretty hard to read.
```suggestion
  config-file.yaml: |
    {{- .Values.kubeRBACProxy.config | toYaml | nindent 4 }}
```

**Review by jkroepke — APPROVED** [review]
LGTM 👍

**Pull request merged**